src/dotnet/commands/dotnet-install/InstallCommand.cs

```diff
@@ -0,0 +1,118 @@
+                packageId,
+                packageVersion,
+                configFile,
+                framework,
+                executablePackagePath);
+
+            DirectoryPath executable = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .WithSubDirectories(
+                    toolConfigurationAndExecutableDirectory
+                        .Configuration
+                        .ToolAssemblyEntryPoint);
+
+            var shellShimMaker = new ShellShimMaker.ShellShimMaker(executablePackagePath.Value);
+            var commandName = toolConfigurationAndExecutableDirectory.Configuration.CommandName;
+            shellShimMaker.EnsureCommandNameUniqueness(commandName);
+
+            shellShimMaker.CreateShim(
+                executable.Value,
+                commandName);
+
+            EnvironmentPathFactory
+                .CreateEnvironmentPathInstruction()
+                .PrintAddPathInstructionIfPathDoesNotExist();
+
+            Reporter.Output.WriteLine(
+                $"{Environment.NewLine}The installation succeeded. If there is no other instruction. You can type the following command in shell directly to invoke: {commandName}");
+
+            return 0;
+        }
+
+        private static ToolConfigurationAndExecutableDirectory ObtainPackage(
+            string packageId,
+            string packageVersion,
+            FilePath? configFile,
+            string framework,
+            DirectoryPath executablePackagePath)
+        {
+            try
+            {
+                var toolPackageObtainer =
+                    new ToolPackageObtainer.ToolPackageObtainer(
+                        executablePackagePath,
+                        () => new DirectoryPath(Path.GetTempPath())
+                            .WithSubDirectories(Path.GetRandomFileName())
+                            .WithFile(Path.GetRandomFileName() + ".csproj"),
+                        new Lazy<string>(BundledTargetFramework.GetTargetFrameworkMoniker),
+                        new PackageToProjectFileAdder(),
+                        new ProjectRestorer());
+
+                return toolPackageObtainer.ObtainAndReturnExecutablePath(
+                    packageId: packageId,
+                    packageVersion: packageVersion,
+                    nugetconfig: configFile,
+                    targetframework: framework);
+            }
+            catch (PackageObtainException ex)
+            {
+                throw new GracefulException(
+                    message:
+                    $"Install failed. Failed to download package:{Environment.NewLine}{ex.Message}",
+                    innerException: ex);
+            }
+            catch (ToolConfigurationException ex)
+            {
+                throw new GracefulException(
+                    message:
+                    $"Install failed. Failed to apply the configuration of the tool:{Environment.NewLine}{ex.Message}",
```

Comment:
@KathleenDollard customer facing string. failed to restore package (package config is wrong)

Reply:
Install failed. The tool's configuration file is not valid. Please contact the owner of the NuGet package. 
<newline>The error was:

* The configuration file is missing
* The configuration file is invalid xml
* The configuration file does not define <an entry point>
* The configuration file contains <an entry point> with invalid characters